scripts/docker-image-builder.sh

```diff
@@ -0,0 +1,102 @@
+#!/usr/bin/env bash
+# 2021-07-08 WATERMARK, DO NOT REMOVE - This script was generated from the Kurtosis Bash script template
+
+set -euo pipefail   # Bash "strict mode"
+
+# ==================================================================================================
+#                                             Main Logic
+# ==================================================================================================
+
+show_helptext_and_exit() {
+    echo "Usage: $(basename "${0}") push_to_registry_container dockerfile_filepath image_tags..."
+    echo ""
+    echo "  push_to_registry_container     Whether images should be pushed to container registry or just loaded locally"
+    echo "  dockerfile_filepath            The absolute path to the Dockerfile to use for this build"
+    echo "  image_tags                     The tags for this image. Multiple values can be passed"
+    echo ""
+    exit 1  # Exit with an error so that if this is accidentally called by CI, the script will fail
+}
+
+# Raw parsing of the arguments
+push_to_registry_container="${1:-}"
+if [ "${push_to_registry_container}" != "true" ] && [ "${push_to_registry_container}" != "false" ]; then
+    echo "Error: Invalid push_to_registry_container arg: '${push_to_registry_container}'" >&2
+    show_helptext_and_exit
+fi
+
+dockerfile_filepath="${2:-}"
+if [ ! -f "${dockerfile_filepath}" ]; then
+    echo "Error: Invalid dockerfile_filepath arg: '${dockerfile_filepath}'" >&2
+    show_helptext_and_exit
+fi
+
+image_tags="${*:3}"
+if [ -z "${image_tags}" ]; then
+    echo "Error: Invalid image_tags arg: '${image_tags}'" >&2
+    show_helptext_and_exit
+fi
+
+# Argument processing
+if "${push_to_registry_container}"; then
+  buildx_platform_arg='linux/arm64/v8,linux/amd64'
+  push_flag='--push'
+else
+  buildx_platform_arg='linux/amd64' # TODO: infer the local arch if that's reasonable
+  push_flag='--load'
+fi
+echo "Building docker image for architecture '${buildx_platform_arg}' with flag '${push_flag}'"
+
+dockerfile_dirpath="$(dirname "${dockerfile_filepath}")"
+echo "Docker file located at '${dockerfile_filepath}' in directory '${dockerfile_dirpath}'"
+
+image_tags_concatenated=""
+for image_tag in ${image_tags}; do
+  echo "Image will be tagged with: '${image_tag}'"
+  image_tags_concatenated=" -t ${image_tag}${image_tags_concatenated} "
+done
+
+# Build Docker image
+kurtosis_docker_builder="kurtosis-docker-builder"
```

Comment:
maybe some formatting can be fixed here and have some new lines between the different blocks!

Reply:
ACK, will add blank lines and some comments